app/src/main/java/com/mmm/parq/activities/HostNewSpotActivity.java

```diff
@@ -0,0 +1,115 @@
+import android.text.Html;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.EditText;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.Volley;
+import com.firebase.client.Firebase;
+import com.google.android.gms.appindexing.AppIndex;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.mmm.parq.R;
+
+import org.json.JSONObject;
+
+public class HostNewSpotActivity extends AppCompatActivity {
+
+    private static final String TAG = HostNewSpotActivity.class.getSimpleName();
+    /**
+     * ATTENTION: This was auto-generated to implement the App Indexing API.
+     * See https://g.co/AppIndexing/AndroidStudio for more information.
+     */
+    private GoogleApiClient mClient;
+    private EditText mAddressField;
+    private EditText mTitleField;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_host_new_spot);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        mAddressField = (EditText) findViewById(R.id.new_spot_address);
+        mTitleField = (EditText) findViewById(R.id.new_spot_title);
+        // ATTENTION: This was auto-generated to implement the App Indexing API.
+        // See https://g.co/AppIndexing/AndroidStudio for more information.
+        mClient = new GoogleApiClient.Builder(this).addApi(AppIndex.API).build();
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.action_finish:
+                Log.i(TAG, "clicked finish");
+                String address = mAddressField.getText().toString();
+                String title = mTitleField.getText().toString();
+                uploadSpot(address, title);
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.host_new_spot_menu, menu);
+        return true;
+    }
+
+    private void uploadSpot(String address, String title) {
+        RequestQueue queue = Volley.newRequestQueue(this);
+        String url = String.format("%s/%s", getString(R.string.api_address), getString(R.string.spots_endpoint));
+        JSONObject data = new JSONObject();
+        try {
+            Firebase ref = new Firebase(getString(R.string.firebase_endpoint));
+            String userId = ref.getAuth().getUid();
+            data.put("userId", userId);
+
+            LocationManager lm = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
```

Comment:
Gotcha -- my only concern is that if we demo adding a spot in class and try to nav to it it might be obvious that we're faking it haha